text/0000-default-field-values.md

```diff
@@ -0,0 +1,1767 @@
+    fn eq(&self, other: &Rhs) -> bool;
+
+    fn ne(&self, other: &Rhs) -> bool { // A default body.
+        !self.eq(other)
+    }
+}
+```
+
+In traits, `const` items can also be assigned a default value. For example:
+
+```rust
+trait Foo {
+    const BAR: usize = 42; // A default value.
+}
+```
+
+Thus, to extend Rust with a notion of field defaults is not an entirely alien
+concept.
+
+## Pattern matching follows construction
+
+[dual]: https://en.wikipedia.org/wiki/Duality_(mathematics)
+
+In mathematics there is a notion of one thing being the *[dual]* of another.
+Loosely speaking, duals are often about inverting something.
+In Rust, one example of such an inversion is expressions and patterns.
+
+Expressions are used to *build up* and patterns *break apart*;
+While it doesn't hold generally, a principle of language design both in Rust
+and other languages with with pattern matching has been that the syntax for
+patterns should, to the extent possible, follow that of expressions.
+
+For example:
+
++ You can match on or build up a struct with `Foo { field }`.
+  For patterns this will make `field` available as a binding
+  while for expressions the binding `field` will be used to build a `Foo`.
+
+  For a tuple struct, `Foo(x)` will work both for construction and matching.
+
++ If you want to be more flexible, both patterns and expressions permit
+  `Foo { field: bar }`.
+
++ You can use both `&x` to dereference and bind to `x` or
+  construct a reference to `x`.
+
++ An array can be constructed with `[a, b, c, d]` and the same is a valid
+  pattern for destructuring an array.
+
+The reason why matching should follow construction is that it makes languages
+easier to understand; you simply learn the expression syntax and then reuse
+it to run the process in reverse.
+
+In some places, Rust could do a better job than it currently does of adhering to
+this principle. In this particular case, the pattern syntax `Foo { a, b: c, .. }`
+has no counterpart in the expression syntax. This RFC rectifies this by
+permitting `Foo { a, b: c, .. }` as an expression syntax; this is identical
+to the expression syntax and thus consistency has been gained.
+
+However, it is not merely sufficient to use the same syntax for expressions;
+the semantics also have to be similar in kind for things to work out well.
+This RFC argues that this is the case because in both contexts, `..` indicates
+something partially ignorable is going on: "I am *destructuring*/*constructing*
+this struct, and by the way there are some more fields I don't care about
+*and let's* drop those* / *and let's fill in with default values*".
+In a way, the use of `_` to mean both a catch-all pattern and type / value
+placeholder is similar to `..`; in the case of `_` both cases indicate something
+unimportant going on. For patterns, `_` matches everything and doesn't give
+access to the value; for types, the placeholder is just an unbounded inference
+variable.
+
+## On privacy
```

Comment:
It would be useful to cross-link to RFC 736 which explains privacy for FRU, and link that here: https://github.com/rust-lang/rfcs/blob/master/text/0736-privacy-respecting-fru.md

Reply:
Added a link to it, explaining how this RFC proposes to somewhat walk it back. Although, I'm not overly concerned with the privacy question, I think there would be value in landing this even if you *can't* construct types with private fields, and relaxing that restriction at a later time if we find it is indeed too restrictive.

Reply:
Changed it so that we don't change the privacy rules, but mention it in future possibilities.